packages/flutter/lib/src/services/text_formatter.dart

```diff
@@ -382,7 +435,12 @@ class LengthLimitingTextInputFormatter extends TextInputFormatter {
         baseOffset: math.min(value.selection.start, truncated.length),
         extentOffset: math.min(value.selection.end, truncated.length),
       ),
-      composing: TextRange.empty,
+      composing: keepComposingRange && truncated.length > value.composing.start && !value.composing.isCollapsed
+        ? TextRange(
+            start: math.min(value.composing.start, truncated.length),
+            end: math.min(value.composing.end, truncated.length),
+          )
```

Comment:
A test case that may fail: `maxLength` = 5, `value.composing` = (5, 6). The function should return a `TextEditingValue` with a composing range of (-1, -1).

Reply:
I'll add a test for it.

Reply:
Actually it won't, `truncated.length > value.composing.start` is false so it goes to empty.

Reply:
oh nice then we don't need the first "math.min"?